### PR TITLE
Loading for search

### PR DESCRIPTION
**Pull request by mruwnik**
https://github.com/StampyAI/stampy-ui/issues/183

When nothing loaded:
image

When questions present:
image

### CODE REVIEW

File: app/server-utils/url-to-iframe.ts [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
+  //   (              <-- capture the url
+  //     https?:\/\/  <-- check for both http and https
+  //     [\.\w]+?     <-- any string of alphanumerical characters and full stops
+  //     \/?[^\]]*    <-- optional query params
+  //   )
+  // \]
+  // \(
+  //     \1           <-- lookback to match whatever the first group caught
+  // \)
+  const linkRegex = new RegExp(/\[(https?:\/\/[.\w]+?\/?[^]]*?)\]\(\1\)/gi)
```
Comment (mruwnik) [inline app/server-utils/url-to-iframe.ts:21]: this wasn't working for me, so I refactored it to work on markdown

Comment (Aprillion) [reply app/server-utils/url-to-iframe.ts:21]: if using regex literal `/.../gi`, the `new RegExp(string)` wrapper is not needed - please choose one of the syntaxes, not both

**Review by Aprillion — APPROVED** [review]

File: app/server-utils/url-to-iframe.ts [hunk 2]
```diff
@@ -1,29 +1,33 @@
 /**
- * Replaces all of the URLs in text with iframes of the URLs
+ * Replaces all whitelisted URLs in the text with iframes of the URLs
```
Comment (Aprillion) [inline app/server-utils/url-to-iframe.ts:2]: (minor) "allowed URLs" would use less jargon for the same explanation

File: app/server-utils/url-to-iframe.ts [hunk 3]
```diff
@@ -1,29 +1,33 @@
...
+  //   (              <-- capture the url
+  //     https?:\/\/  <-- check for both http and https
+  //     [\.\w]+?     <-- any string of alphanumerical characters and full stops
+  //     \/?[^\]]*    <-- optional query params
+  //   )
+  // \]
+  // \(
+  //     \1           <-- lookback to match whatever the first group caught
+  // \)
+  const matches = text.matchAll(/\[(https?:\/\/[.\w]+?\/?[^]]*?)\]\(\1\)/gi)
```
Comment (Aprillion) [inline app/server-utils/url-to-iframe.ts:21]: do we want to replace only links in the `[url](url)` format on purpose, keeping `[text](url)` as text links?

**Review by Aprillion — APPROVED** [review]

File: app/server-utils/url-to-iframe.ts (hunk 3 above)
Comment (mruwnik) [reply app/server-utils/url-to-iframe.ts:21]: yes. The idea is that there are a whitelisted collection of urls that are to be transformed into iframes, but only if both the url and text are the same. So you could do `[here's a map](http://aisafety.world/)` in the main text, without it then being transformed into an iframe of the map

**Pull request merged**